File: NotesApp/NotesApp/Models/Tag.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace NotesApp.Models;

public class Tag
{
    public const string DefaultColorHex = "#4C8DFF";

    [Key]
    public int Id { get; set; }

    [Required]
    [MaxLength(50)]
    public string Name { get; set; } = string.Empty;

    [Required]
    [MaxLength(9)]
    public string ColorHex { get; set; } = DefaultColorHex;

    public ICollection<NoteTag> NoteTags { get; set; } = new List<NoteTag>();
}

File: Sources/NotesCore/Models/Tag.swift
import Foundation

public struct Tag: Identifiable, Hashable {
    public static let defaultColorHex = "#4C8DFF"

    public var id: Int64
    public var name: String
    public var colorHex: String

    public init(id: Int64, name: String, colorHex: String = Tag.defaultColorHex) {
        self.id = id
        self.name = name
        self.colorHex = colorHex
    }
}
